addons/purchase_stock/models/product.py

```diff
@@ -43,15 +43,15 @@ def _get_quantity_in_progress(self, location_ids=False, warehouse_ids=False):
                 '|',
                     ('order_id.picking_type_id.default_location_dest_id', 'in', location_ids),
                     '&',
-                        ('move_dest_ids', '=', False),
+                        ('move_dest_bool', '=', False),
```

Comment:
could you move whole domain to a new function _get_purchase_lines_domain(location_ids, warehouse_ids)
that way we will be able to make a back port of this  as new module.


With the current approach, a new module holding this patch won't solve the issue, since the super gonna call at the end the buggy domain using the move_dest_ids

Reply:
Hi Hugo,

I understand that having a new function to get domain would be a perfect solution, but I have to refuse the idea.
We must to make minimal changes in stable version and avoid extra risk. Particularly we avoid any refactoring in non-master branches. 

You sill can fix the performance issue via custom module: you should copy-paste method `_get_quantity_in_progress` and replace the line with super() call to the line from  stock module

https://github.com/odoo/odoo/blob/5b335396addb16456d61273ae6b0446913f7f489/addons/stock/models/product.py#L592-L593

i.e.

``` diff
-        qty_by_product_location, qty_by_product_wh = super()._get_quantity_in_progress(location_ids, warehouse_ids)
+        qty_by_product_location, qty_by_product_wh = defaultdict(float), defaultdict(float)

Reply:
Hi Hugo,

I understand that having a new function to get domain would be a perfect solution, but I have to refuse the idea.
We must to make minimal changes in stable version and avoid extra risk. Particularly we avoid any refactoring in non-master branches. 

You sill can fix the performance issue via custom module: you should copy-paste method `_get_quantity_in_progress` and replace the line with super() call to the line from  stock module

https://github.com/odoo/odoo/blob/5b335396addb16456d61273ae6b0446913f7f489/addons/stock/models/product.py#L592-L593

i.e.

``` diff
-        qty_by_product_location, qty_by_product_wh = super()._get_quantity_in_progress(location_ids, warehouse_ids)
+        qty_by_product_location, qty_by_product_wh = defaultdict(float), defaultdict(float)

Reply:
@yelizariev 

It was already split in stable branch

could you use it pls?

https://github.com/odoo/odoo/commit/cb9314e902fc899c5ab42b6fd4d8c1ed811c24e7

Reply:
@hugho-ad done

Thanks for your assistance with refactoring :wink:

Reply:
Awesome, thanks to you man!